### PR TITLE
[ROCm] enable gfx110x for hipblaslt

### LINKED ISSUE #136347: Compile hipBLASlt to support gfx1100
### 🚀 The feature, motivation and pitch

According to [this post](https://github.com/pytorch/torchtune/discussions/1108#discussioncomment-9851037), it's just a matter of compiling for the gfx1100 architecture and it will work. Supposedly the gfx1100 architecture also exists in the `CMakeLists.txt` file in the upstream sources, so it should be readily supported.

### Alternatives

_No response_

### Additional context

_No response_

cc @malfet @seemethere @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

### PR DESCRIPTION
**Pull request by jeffdaily**
Fixes #136347.

cc @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @naromero77amd

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: aten/src/ATen/Context.cpp [hunk 1]
```diff
@@ -290,7 +290,11 @@ at::BlasBackend Context::blasPreferredBackend() {
 #ifdef USE_ROCM
   if (blas_preferred_backend == at::BlasBackend::Cublaslt) {
     static const bool hipblaslt_unsupported = []() {
+#if ROCM_VERSION >= 60200
+      static const std::vector<std::string> archs = {"gfx90a", "gfx940", "gfx941", "gfx942", "gfx1100", "gfx1101"};
+#else
       static const std::vector<std::string> archs = {"gfx90a", "gfx940", "gfx941", "gfx942"};
+#endif
```
Comment (jithunnair-amd) [inline aten/src/ATen/Context.cpp:297]: Could remove some repetition?

```suggestion
    static const std::vector<std::string> archs = {"gfx90a", "gfx940", "gfx941", "gfx942", 
#if ROCM_VERSION >= 60200
    "gfx1100", "gfx1101",
#endif
    };
```

Comment (jeffdaily) [reply aten/src/ATen/Context.cpp]: Great suggestion.  Made the same change in both places.

**Comment by naromero77amd** [comment]
@jeffdaily I can get the patch to work as long as I do: 
```
export TORCH_BLAS_PREFER_HIPBLASLT=1
```

**Comment by emaibb** [comment]
@eqy @syed-ahmed Please merge it. Just a tiny change, then gfx11 is supported.

**Comment by naromero77amd** [comment]
> @jeffdaily I can get the patch to work as long as I do:
> 
> ```
> export TORCH_BLAS_PREFER_HIPBLASLT=1
> ```
We did some further testing yesterday with ROCm 6.2 and found some problems. Please do **not** merge until we can take a closer look at what is going on.

**Comment by naromero77amd** [comment]
hipBLASLt was missing support for float32 GEMM operations in ROCm 6.2. The change is pushed out to ROCm 6.3.

**Comment by naromero77amd** [comment]
@jithunnair-amd This one is ready for re-review as well. (I don't have permission to take it out of draft mode).

**Review by jithunnair-amd — APPROVED** [review]

**Comment by jithunnair-amd** [comment]
@jeffdaily This PR cannot be tested as of now in upstream CI, your call on merge.

**Pull request closed without merging**